packages/client/src/links/httpBatchLink/streamingHttpUtils.ts

```diff
@@ -0,0 +1,163 @@
+// Stream parsing adapted from https://www.loginradius.com/blog/engineering/guest-post/http-streaming-with-nodejs-and-fetch-api/
+
+import { TRPCResponse } from '@trpc/server/rpc';
+import { TRPCClientError } from '../../TRPCClientError';
+import {
+  HTTPBaseRequestOptions,
+  HTTPResult,
+  fetchHTTPResponse,
+  getBody,
+  getUrl,
+} from '../internals/httpUtils';
+import { HTTPHeaders } from '../types';
+
+/**
+ * @param readableStream as given by `(await fetch(url)).body`
+ * @param onSingle called for each line of the stream
+ * @param parser defaults to `JSON.parse`
+ */
+export async function parseJsonStream<TReturn>(
```

Comment:
I assume all this works correctly but let's make sure we have a plan in place to maintain/change this code if necessary

Reply:
What do you mean?